From ce7620ac558c80709628405d8f3d5472b5fc0438 Mon Sep 17 00:00:00 2001
From: "Ryan McCabe Date: Tue, 13 Feb 2018 22:42:17 -0500" <rmccabe@redhat.com>
Date: Thu, 14 Jun 2018 15:54:59 +0000
Subject: [PATCH] azure: Fix publishing of hostname

Set the DHCP_HOSTNAME env var before bouncing the interfaces
to ensure that the hostname is published correctly.

Resolves: rhbz#1434109

X-downstream-only: yes
Signed-off-by: Ryan McCabe <rmccabe@redhat.com>
---
 cloudinit/sources/DataSourceAzure.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/cloudinit/sources/DataSourceAzure.py b/cloudinit/sources/DataSourceAzure.py
index 0ee622e..517baa7 100644
--- a/cloudinit/sources/DataSourceAzure.py
+++ b/cloudinit/sources/DataSourceAzure.py
@@ -31,7 +31,8 @@ AGENT_START = ['service', 'walinuxagent', 'start']
 AGENT_START_BUILTIN = "__builtin__"
 BOUNCE_COMMAND_IFUP = [
     'sh', '-xc',
-    "i=$interface; x=0; ifdown $i || x=$?; ifup $i || x=$?; exit $x"
+    "i=$interface; DHCP_HOSTNAME=$hostname; x=0; "
+    "ifdown $i || x=$?; ifup $i || x=$?; exit $x"
 ]
 BOUNCE_COMMAND_FREEBSD = [
     'sh', '-xc',
